qep-345-line-symbollayer-blankareas.md

```diff
@@ -0,0 +1,117 @@
+# QGIS Enhancement: Markers/Hashes line symbol layer blank areas 
+
+**Date** 2025/07/22
+
+**Author** Julien Cabieces (@troopa81)
+
+**Contact** julien dot cabieces at oslandia dot com
+
+**Version** QGIS 4.0
+
+# Summary
+
+This proposal relates to the line symbol layer which draw markers or hashes at intervals along a line. When using such symbol layer, the display of the repeated pattern (hash or marker) can be ugly, and user would like to have a better control on what's displayed.
+
+![issue.png](images/qep345/issue.png)
+
+The rest of the proposal will mention markers but everything describe here applies also to hashes.
+
+## Proposed Solution
+
+In order to have a proper display of the markers along the line, I propose to:
+- Give user a way to identify blank area where markers are not displayed
+- Use the already existing annotation feature to add, move, rotate or even change marker symbology
```

Comment:
There's absolutely no detail about annotations in this QEP. Is this section outdated? Or is this QEP incomplete?

Reply:
No there is nothing missing. I just wanted to bring some context to explain that user can already use annotations to place marker wherever he wants. So this QEP is only about blank areas, nothing related to annotation.

Reply:
Can you drop this bit? I think it's misleading to include if you're not actually going to implement any changes there. 

(FWIW, I don't think annotations are a good solution for manual hash/marker placement -- there'd be no way to link the symbology of the original line to the custom annotation marker/hashes, so any need to change colors/line thickness/marker symbol would be a horrible painful experience requiring changes to many individual annotation objects as well as the original layer :scream: )

Reply:
> (FWIW, I don't think annotations are a good solution for manual hash/marker placement -- there'd be no way to link the symbology of the original line to the custom annotation marker/hashes, so any need to change colors/line thickness/marker symbol would be a horrible painful experience requiring changes to many individual annotation objects as well as the original layer 😱 )

True. 

I replaced this bit with 

> User could then use another point layer with marker symbol to place and rotate other marker where he wants to. 

I don't want to add too much specific to marker line symbol layer (like adding, moving) and would like to rely on what already existing.

Reply:
Since this QEP is about blank areas only, this is out of scope, but still out of curiosity: What difference does it make when it's a point layer? It has neither a link to the original geometry, right?